File: src/ros2_bridge/mappers/geometry_msgs/polygon_instance_stamped.rs
//! Typed mapper for `geometry_msgs/msg/PolygonInstanceStamped`.

use crate::ros2_bridge::mapper::TypedTopicMapper;

pub(crate) fn polygon_instance_stamped_to_bus(msg: ros_env::geometry_msgs::msg::PolygonInstanceStamped) -> crate::geometry_msgs::msg::v1::PolygonInstanceStamped {
    crate::geometry_msgs::msg::v1::PolygonInstanceStamped {
        header: Some(crate::ros2_bridge::mappers::std_msgs::header::header_to_bus(msg.header)),
        polygon: Some(crate::ros2_bridge::mappers::geometry_msgs::polygon_instance::polygon_instance_to_bus(msg.polygon)),
    }
}

pub(crate) fn polygon_instance_stamped_to_ros(bus: crate::geometry_msgs::msg::v1::PolygonInstanceStamped) -> ros_env::geometry_msgs::msg::PolygonInstanceStamped {
    ros_env::geometry_msgs::msg::PolygonInstanceStamped {
        header: crate::ros2_bridge::mappers::std_msgs::header::header_to_ros(bus.header.unwrap_or_default()),
        polygon: crate::ros2_bridge::mappers::geometry_msgs::polygon_instance::polygon_instance_to_ros(bus.polygon.unwrap_or_default()),
    }
}

#[derive(Clone, Copy, Debug, Default)]
pub struct GeometryMsgsPolygonInstanceStampedMapper;

impl TypedTopicMapper for GeometryMsgsPolygonInstanceStampedMapper {
    type Ros = ros_env::geometry_msgs::msg::PolygonInstanceStamped;
    type Bus = crate::geometry_msgs::msg::v1::PolygonInstanceStamped;

    fn type_name(&self) -> &'static str {
        "geometry_msgs/msg/PolygonInstanceStamped"
    }

    fn ros_to_bus(&self, msg: Self::Ros) -> crate::errors::Result<Self::Bus> {
        Ok(polygon_instance_stamped_to_bus(msg))
    }

    fn bus_to_ros(&self, msg: Self::Bus) -> crate::errors::Result<Self::Ros> {
        Ok(polygon_instance_stamped_to_ros(msg))
    }
}

File: bindings/python/robot_bus/ros2_bridge/mappers/geometry_msgs/polygon_instance_stamped.py
"""Generated mapper for `geometry_msgs/msg/PolygonInstanceStamped`."""

from __future__ import annotations

from robot_bus.ros2_bridge.mappers import _convert
from robot_bus.ros2_bridge.mappers.std_msgs.header import header_to_bus, header_to_ros
from robot_bus.ros2_bridge.mappers.geometry_msgs.polygon_instance import polygon_instance_to_bus, polygon_instance_to_ros

_BusMsg = None


def _bus_cls():
    global _BusMsg
    if _BusMsg is None:
        from robot_bus.geometry_msgs.msg.v1 import PolygonInstanceStamped as BusMsg

        _BusMsg = BusMsg
    return _BusMsg


def polygon_instance_stamped_to_bus(msg):
    BusMsg = _bus_cls()
    bus = BusMsg()
    bus.header.CopyFrom(header_to_bus(msg.header))
    bus.polygon.CopyFrom(polygon_instance_to_bus(msg.polygon))
    return bus


def polygon_instance_stamped_to_ros(bus):
    from geometry_msgs.msg import PolygonInstanceStamped as RosMsg

    out = RosMsg()
    out.header = header_to_ros(bus.header)
    out.polygon = polygon_instance_to_ros(bus.polygon)
    return out


class GeometryMsgsPolygonInstanceStampedMapper:
    _ros_type = None

    def ros_msg_type(self):
        cls = type(self)
        if cls._ros_type is None:
            from geometry_msgs.msg import PolygonInstanceStamped as RosMsg

            cls._ros_type = RosMsg
        return cls._ros_type

    def ros_to_bus(self, msg) -> bytes:
        return polygon_instance_stamped_to_bus(msg).SerializeToString()

    def bus_to_ros(self, payload: bytes):
        BusMsg = _bus_cls()
        bus = BusMsg()
        bus.ParseFromString(payload)
        return polygon_instance_stamped_to_ros(bus)
